awx/ui_next/src/components/Lookup/CredentialsLookup.jsx

```diff
@@ -0,0 +1,165 @@
+  handleCredentialTypeSelect(value, type) {
+    const {credentialTypes} = this.state
+    const selectedType = credentialTypes.filter(item => item.label === type)
+    this.setState({selectedCredentialType: selectedType[0]})
+  }
+
+  toggleCredential(item) {
+    const { credentials: stateToUpdate, selectedCredentialType } = this.state;
+    const { onChange } = this.props;
+    const index = stateToUpdate.findIndex(
+      credential => credential.id === item.id
+    )
+    if (index > -1) {
+      const newCredentialsList = stateToUpdate.filter(cred => cred.id !== item.id)
+      this.setState({ credentials: newCredentialsList });
+      onChange(newCredentialsList)
+      return;
+    }
+
+    const credentialTypeOccupied = stateToUpdate.some(cred => cred.kind === item.kind);
+    if (selectedCredentialType.value === "Vault" || !credentialTypeOccupied ) {
+      item.credentialType = selectedCredentialType
+      this.setState({ credentials: [...stateToUpdate, item] })
+      onChange([...stateToUpdate, item])
+    } else {
+      const credsList = [...stateToUpdate]
+      const occupyingCredIndex = stateToUpdate.findIndex(occupyingCred =>
+        occupyingCred.kind === item.kind)
+      credsList.splice(occupyingCredIndex, 1, item)
+      this.setState({ credentials: credsList })
+      onChange(credsList)
+    }
+  }
+
+  render() {
+    const { tooltip, i18n } = this.props;
+    const { credentials, selectedCredentialType, credentialTypes } = this.state;
+
+    return (
+        <FormGroup
+          label={i18n._(t`Credentials`)}
+          fieldId="org-credentials"
+        >
+          {tooltip && (
+            <Tooltip position="right" content={tooltip}>
+              <QuestionCircleIcon />
+            </Tooltip>
+          )}
+          {credentialTypes && (
+            <Lookup
+            selectCategoryOptions={credentialTypes}
+            selectCategory={this.handleCredentialTypeSelect}
+            selectedCategory={selectedCredentialType}
+            onToggleItem = {this.toggleCredential}
+            onloadCategories={this.loadCredentialTypes}
+            id="org-credentials"
+            lookupHeader={i18n._(t`Credentials`)}
+            name="credentials"
+            value={credentials}
+            multiple
+            onLookupSave={() => {}}
+            getItems={this.loadCredentials}
+            qsNamespace="credentials"
+            columns={[
+              {
+                name: i18n._(t`Name`),
+                key: 'name',
+                isSortable: true,
+                isSearchable: true,
+              },
+            ]}
+            sortedColumnKey="name"
+          />
+          )}
+        </FormGroup>
+    );
+  }
+}
+
+CredentialsLookup.propTypes = {
+  tooltip: PropTypes.string,
+  onChange: PropTypes.func.isRequired,
+};
+
+CredentialsLookup.defaultProps = {
+  tooltip: '',
+};
+export { CredentialsLookup as _CredentialsLookup };
+
+export default withI18n()(withRouter(CredentialsLookup));
```

Comment:
Oof.  Probably don't need to export `withRouter` here.  Will remove after getting feedback from others.